Test failing with libarchive 3.2

Since libarchive 3.2, the test test_check_archiveentry_with_unicode_and_binary_entries_zip2 is failing:

```
collected 16 items 

tests/test_convert.py::test_convert PASSED
tests/test_entry.py::test_entry_properties PASSED
tests/test_entry.py::test_check_ArchiveEntry_against_TarInfo PASSED
tests/test_entry.py::test_check_archiveentry_using_python_testtar PASSED
tests/test_entry.py::test_check_archiveentry_with_unicode_and_binary_entries_tar PASSED
tests/test_entry.py::test_check_archiveentry_with_unicode_and_binary_entries_zip PASSED
tests/test_entry.py::test_check_archiveentry_with_unicode_and_binary_entries_zip2 FAILED
tests/test_entry.py::test_check_archiveentry_with_unicode_entries_and_name_zip PASSED
tests/test_errors.py::test_add_files_nonexistent PASSED
tests/test_errors.py::test_check_int_logs_warnings PASSED
tests/test_errors.py::test_check_null PASSED
tests/test_errors.py::test_error_string_decoding PASSED
tests/test_rwx.py::test_buffers PASSED
tests/test_rwx.py::test_fd PASSED
tests/test_rwx.py::test_files PASSED
tests/test_rwx.py::test_custom_writer PASSED

================================================================================================================== FAILURES ===================================================================================================================
________________________________________________________________________________________ test_check_archiveentry_with_unicode_and_binary_entries_zip2 _________________________________________________________________________________________

    def test_check_archiveentry_with_unicode_and_binary_entries_zip2():
>       check_entries(join(data_dir, 'unicode2.zip'))

tests/test_entry.py:76: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

test_file = '/tmp/debian/archivec/python-libarchive-c-2.1/tests/data/unicode2.zip', regen = False

    def check_entries(test_file, regen=False):
        fixture_file = test_file + '.json'
        if regen:
            entries = list(get_entries(test_file))
            with open(fixture_file, 'w', encoding='UTF-8') as ex:
                json.dump(entries, ex, indent=2)
        with open(fixture_file, encoding='UTF-8') as ex:
            expected = json.load(ex)
        actual = list(get_entries(test_file))
        for e1, e2 in zip(actual, expected):
>           assert e1 == e2
E           assert {'isblk': Fal...r': True, ...} == {'isblk': Fals...r': True, ...}
E             Common items:
E             {'isblk': False,
E              'ischr': False,
E              'isdev': False,
E              'isdir': True,
E              'isfifo': False,
E              'islnk': False,
E              'isreg': False,
E              'issym': False,
E              'linkpath': None,
E              'mtime': 1381752672,
E              'path': 'a/',
E              'size': 0}
E             Differing items:
E             {'mode': 'rwxrwxr-x'} != {'mode': 'rwxrwxrwx'}
E             Full diff:
E             {'isblk': False,
E             'ischr': False,
E             'isdev': False,
E             'isdir': True,
E             'isfifo': False,
E             'islnk': False,
E             'isreg': False,
E             'issym': False,
E             'linkpath': None,
E             -  'mode': 'rwxrwxr-x',
E             ?                  ^
E             +  'mode': 'rwxrwxrwx',
E             ?                  ^
E             'mtime': 1381752672,
E             'path': 'a/',
E             'size': 0}

tests/test_entry.py:93: AssertionError
===================================================================================================== 1 failed, 15 passed in 0.22 seconds =====================================================================================================
```

--- a/tests/data/unicode2.zip.json
+++ b/tests/data/unicode2.zip.json
@@ -10,7 +10,7 @@
     "isblk": false, 
     "isreg": false, 
     "isfifo": false, 
-    "mode": "rwxrwxrwx", 
+    "mode": null, 
     "islnk": false, 
     "ischr": false
   }, 
@@ -25,7 +25,7 @@
     "isblk": false, 
     "isreg": true, 
     "isfifo": false, 
-    "mode": "rw-rw-rw-", 
+    "mode": null, 
     "islnk": false, 
     "ischr": false
   }

--- a/tests/test_entry.py
+++ b/tests/test_entry.py
@@ -68,14 +68,15 @@ def test_check_archiveentry_with_unicode_and_binary_entries_zip():
 
 
 def test_check_archiveentry_with_unicode_and_binary_entries_zip2():
-    check_entries(join(data_dir, 'unicode2.zip'))
+    check_entries(join(data_dir, 'unicode2.zip'), ignore='mode')
 
 
 def test_check_archiveentry_with_unicode_entries_and_name_zip():
     check_entries(join(data_dir, '\ud504\ub85c\uadf8\ub7a8.zip'))
 
 
-def check_entries(test_file, regen=False):
+def check_entries(test_file, regen=False, ignore=''):
+    ignore = ignore.split()
     fixture_file = test_file + '.json'
     if regen:
         entries = list(get_entries(test_file))
@@ -85,4 +86,7 @@ def check_entries(test_file, regen=False):
         expected = json.load(ex)
     actual = list(get_entries(test_file))
     for e1, e2 in zip(actual, expected):
+        for key in ignore:
+            e1.pop(key)
+            e2.pop(key)
         assert e1 == e2

